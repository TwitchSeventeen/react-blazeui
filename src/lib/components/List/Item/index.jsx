import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Item(props) {
  const {
    children, className, unstyled, dataName, ...other
  } = props;
  const classes = classNames(
    'c-list__item',
    { 'c-list__item--unstyled': unstyled },
    className,
  );

  return (
    <li
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </li>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines an unstyled list
   */
  unstyled: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Item.defaultProps = {
  children: null,
  className: null,
  unstyled: false,
  dataName: 'list-item',
};

export default Item;
