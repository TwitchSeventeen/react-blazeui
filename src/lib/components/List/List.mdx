---
name: List
menu: Components
---

import { Playground, PropsTable } from 'docz'
import List from './index';

# List

Ordered, un-ordered, indented, you're covered when it comes to making lists.

<Playground>
    <List>
        <List.Item>Lorem ipsum dolor sit amet</List.Item>
        <List.Item>Consectetur adipiscing elit</List.Item>
        <List.Item>Nulla volutpat aliquam velit
            <List>
                <List.Item>Phasellus iaculis neque</List.Item>
                <List.Item>Purus sodales ultricies</List.Item>
                <List.Item>Vestibulum laoreet porttitor sem</List.Item>
                <List.Item>Ac tristique libero volutpat at</List.Item>
            </List>
        </List.Item>
        <List.Item>Faucibus porta lacus fringilla vel</List.Item>
    </List>
</Playground>

## Ordered
The **ordered** prop will apply correct numbering to ordered lists and nested lists.

<Playground>
    <List ordered>
        <List.Item>Lorem ipsum dolor sit amet</List.Item>
        <List.Item>Consectetur adipiscing elit</List.Item>
        <List.Item>Nulla volutpat aliquam velit
            <List ordered>
                <List.Item>Phasellus iaculis neque</List.Item>
                <List.Item>Purus sodales ultricies
                    <List ordered>
                        <List.Item>Lorem ipsum dolor sit amet</List.Item>
                        <List.Item>Consectetur adipiscing elit</List.Item>
                    </List>
                </List.Item>
                <List.Item>Vestibulum laoreet porttitor sem</List.Item>
                <List.Item>Ac tristique libero volutpat at</List.Item>
            </List>
        </List.Item>
        <List.Item>Faucibus porta lacus fringilla vel</List.Item>
    </List>
</Playground>

## Un-styled
Applying the **unstyled** prop to a List will remove the bullet points.

You can also remove bullet points on individual Items by adding the **unstyled** prop.

<Playground>
    <List unstyled>
        <List.Item>This list is unstyled</List.Item>
        <List.Item>Consectetur adipiscing elit</List.Item>
        <List.Item>Nulla volutpat aliquam velit
            <List>
                <List.Item>Phasellus iaculis neque</List.Item>
                <List.Item>Purus sodales ultricies</List.Item>
                <List.Item unstyled>This is an unstyled item</List.Item>
                <List.Item>Ac tristique libero volutpat at</List.Item>
            </List>
        </List.Item>
        <List.Item>Faucibus porta lacus fringilla vel</List.Item>
    </List>
</Playground>

## Inline
To make your list items flow left to right apply the **inline** prop.

<Playground>
    <List inline>
        <List.Item>Lorem</List.Item>
        <List.Item>ipsum</List.Item>
        <List.Item>dolor</List.Item>
    </List>
</Playground>

## Properties
<PropsTable of={List} />