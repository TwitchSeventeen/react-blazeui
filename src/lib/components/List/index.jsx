import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Item from './Item';

function List(props) {
  const {
    children, className, ordered, unstyled, inline,
  } = props;
  const classes = classNames(
    'c-list',
    { 'c-list--ordered': ordered },
    { 'c-list--unstyled': unstyled },
    { 'c-list--inline': inline },
    className,
  );

  if (ordered) {
    return (
      <ol {...props} className={classes}>
        { children }
      </ol>
    );
  }

  return (
    <ul {...props} className={classes}>
      { children }
    </ul>
  );
}

List.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines an ordered list
   */
  ordered: PropTypes.bool,
  /**
   * Defines an unstyled list
   */
  unstyled: PropTypes.bool,
  /**
   * Defines an inline list
   */
  inline: PropTypes.bool,
};

List.defaultProps = {
  children: null,
  className: null,
  ordered: false,
  unstyled: false,
  inline: false,
};


List.Item = Item;

export default List;
