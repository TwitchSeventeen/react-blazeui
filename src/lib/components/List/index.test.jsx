import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import List from './index';

describe('List', () => {
  it('should be defined', () => {
    expect(List).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <List>
          <List.Item>lorem</List.Item>
          <List.Item>ipsum</List.Item>
          <List.Item>dolor</List.Item>
        </List>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly an ordered list', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <List ordered>
          <List.Item>lorem</List.Item>
          <List.Item>ipsum</List.Item>
          <List.Item>dolor</List.Item>
        </List>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly an unstyled list', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <List unstyled>
          <List.Item>lorem</List.Item>
          <List.Item>ipsum</List.Item>
          <List.Item>dolor</List.Item>
        </List>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly an inline list', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <List inline>
          <List.Item>lorem</List.Item>
          <List.Item>ipsum</List.Item>
          <List.Item>dolor</List.Item>
        </List>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
