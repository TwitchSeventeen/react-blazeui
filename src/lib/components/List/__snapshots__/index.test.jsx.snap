// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<ul
  className="c-list"
>
  <Item
    className={null}
    unstyled={false}
  >
    lorem
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    ipsum
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    dolor
  </Item>
</ul>
`;

exports[`List renders correctly an inline list 1`] = `
<ul
  className="c-list c-list--inline"
>
  <Item
    className={null}
    unstyled={false}
  >
    lorem
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    ipsum
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    dolor
  </Item>
</ul>
`;

exports[`List renders correctly an ordered list 1`] = `
<ol
  className="c-list c-list--ordered"
>
  <Item
    className={null}
    unstyled={false}
  >
    lorem
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    ipsum
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    dolor
  </Item>
</ol>
`;

exports[`List renders correctly an unstyled list 1`] = `
<ul
  className="c-list c-list--unstyled"
>
  <Item
    className={null}
    unstyled={false}
  >
    lorem
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    ipsum
  </Item>
  <Item
    className={null}
    unstyled={false}
  >
    dolor
  </Item>
</ul>
`;
